frontend/packages/client/src/components/ProposalCreate/StepOne/index.js

```diff
@@ -242,27 +262,64 @@ const StepOne = ({
       })
       .getLastCreatedEntityKey();
 
-    const newEditorStateWidthImage = AtomicBlockUtils.insertAtomicBlock(
+    const newESWidthImageAndExtraBlock = AtomicBlockUtils.insertAtomicBlock(
       editorState,
       entityKey,
       " "
     );
+    // user did not add caption text
+    if (caption.length === 0) {
+      return newESWidthImageAndExtraBlock;
+    }
+    // using cs: content state
+    const contentState = newESWidthImageAndExtraBlock.getCurrentContent();
+
+    const lastBlockAdded = contentState.getLastBlock();
+    const lastBlockAddedKey = lastBlockAdded.getKey();
+
+    // filter and remove the last block added bc it's not necessary
+    const contentStateUpdated = ContentState.createFromBlockArray(
+      contentState
+        .getBlocksAsArray()
+        .filter((block) => block.getKey() !== lastBlockAddedKey),
+      contentState.getEntityMap()
+    );
+
+    // get existing blocks
+    const blockMapArray = contentStateUpdated.getBlocksAsArray();
 
-    const contentState = newEditorStateWidthImage.getCurrentContent();
-    const selectionState = newEditorStateWidthImage.getSelection();
+    // create new temporal content state to extract block with text
+    const tempCSWithCaption = ContentState.createFromText(caption);
+    // get the block with the text from temp content
+    const [tempBlockArray] = tempCSWithCaption.getBlocksAsArray();
 
-    const contentStateWithCaption = Modifier.insertText(
-      contentState,
-      selectionState,
-      caption,
-      ["IMAGE_CAPTION"]
+    // update block type so it's a custom type: image-caption
+    const csWithUpdatedBlock = Modifier.setBlockType(
+      ContentState.createFromBlockArray([tempBlockArray]),
+      SelectionState.createEmpty(tempBlockArray.key),
+      "image-caption-block"
     );
+    // get the block with custom type and with text
+    const [updatedBlock] = csWithUpdatedBlock.getBlocksAsArray();
 
-    const newEditorState = EditorState.set(newEditorStateWidthImage, {
-      currentContent: contentStateWithCaption,
-    });
+    // add block updated and concat empty block at the end
+    const newContentState = ContentState.createFromBlockArray(
+      blockMapArray.concat(updatedBlock).concat(lastBlockAdded),
```

Comment:
instead of chaining methods I think you could just use `.concat(updatedBlock, lastBlockAdded)`

Reply:
updated